### PR TITLE
Make owner optional in experiment POST endpoint.

### LINKED ISSUE #1928: [Feature] Update Experiment REST API POST endpoint to make owner optional
### Description of Feature

We want to support owner-less experiments. Currently, the `owner` property is required.

With this update, the `owner` field should be optional. If it's not provided, we can skip it, but if it's provided, we use the current flow where we look up the user by email address.

Additionally, if the auth was done via a personal access token, and an owner property wasn't provided, we should default to the owner of the personal access token.

### PR DESCRIPTION
**Pull request by romain-growthbook**
### Features and Changes

This PR addresses the changes described in https://github.com/growthbook/growthbook/issues/1928. It makes `owner` body argument optional and uses the request's `userId` for its value when available and no `owner` has been spericified.

- Closes #1928

### CODE REVIEW

File: packages/back-end/src/api/experiments/postExperiment.ts [hunk 1]
```diff
@@ -47,20 +47,28 @@ export const postExperiment = createApiRequestHandler(postExperimentValidator)(
       );
     }
 
-    const user = await getUserByEmail(owner);
+    const ownerPayload = await (async () => {
+      if (!ownerParam) {
+        return req.context.userId ? { owner: req.context.userId } : {};
```
Comment (romain-growthbook) [inline packages/back-end/src/api/experiments/postExperiment.ts:52]: The original feature request specification is:
> if the auth was done via a personal access token, and an owner property wasn't provided, we should default to the owner of the personal access token.

Is that the right interpretation?

File: packages/back-end/src/api/experiments/postExperiment.ts [hunk 2]
```diff
@@ -47,20 +47,26 @@ export const postExperiment = createApiRequestHandler(postExperimentValidator)(
       );
     }
 
-    const user = await getUserByEmail(owner);
+    const ownerId = await (async () => {
+      if (!onwerEmail) return req.context.userId;
```
Comment (romain-growthbook) [inline packages/back-end/src/api/experiments/postExperiment.ts:51]: The original feature request specification is:
> if the auth was done via a personal access token, and an owner property wasn't provided, we should default to the owner of the personal access token.

Is that the right interpretation?

Comment (jdorn) [reply packages/back-end/src/api/experiments/postExperiment.ts:51]: Yes, that's correct

File: packages/back-end/src/api/experiments/postExperiment.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ export const postExperiment = createApiRequestHandler(postExperimentValidator)(
   async (req): Promise<PostExperimentResponse> => {
     req.checkPermissions("createAnalyses", req.body.project);
 
-    const { datasourceId, owner } = req.body;
+    const { datasourceId, owner: onwerEmail } = req.body;
```
Comment (jdorn) [inline packages/back-end/src/api/experiments/postExperiment.ts:20]: Typo: `onwer` instead of `owner`

Comment (romain-growthbook) [reply packages/back-end/src/api/experiments/postExperiment.ts]: thanks, fixed.

**Review by jdorn — APPROVED** [review]

**Pull request merged**